src/main/resources/services/collectorAssets/collectorAssets.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+	Controller,
+	Request,
+	ResourceKey
+} from '../../../../../../xp-comlock-9742/modules/lib/core/index';
+
+import lcKeys from '@enonic/js-utils/object/lcKeys';
+import {startsWith} from '@enonic/js-utils/string/startsWith';
+// @ts-expect-error no types
+import Router from '/lib/router';
+import {getResource} from '/lib/xp/io';
+
+
+const appHelper = __.newBean<{
+	toResourceKey(resourceUri: string): ResourceKey
+}>('com.enonic.app.explorer.AppHelper');
+
+const etagService = __.newBean<{
+	getEtag: (path: string, etagOverride?: number) => Record<string,string>
+}>('lib.enonic.libStatic.etag.EtagService');
```

Comment:
It is a bad idea to use Java classes from libs.
Java classes are no lib apis - their stability or even existence in the next bugfix version is not guaranteed.
Copy-paste the entire thing instead.

Reply:
Right so we should export some js function in lib-static to generate etags for resources.

Reply:
it fits better in lib-io. But this is another discussion.
I don't think that lib-static should be able to sneak into resources of other apps.

Reply:
Since our webpage (https://www.enonic.com/platform/overview) says our platform provides a JavaScript API , I'm happy everytime I get some actual JavaScript API :)

Reply:
Java classes copied over